layers.go

```diff
@@ -610,6 +614,58 @@ func (r *layerStore) Status() ([][2]string, error) {
 	return r.driver.Status(), nil
 }
 
+func (r *layerStore) PutAdditionalLayer(id string, parentLayer *Layer, names []string, aLayer drivers.AdditionalLayer) (layer *Layer, err error) {
+	if duplicateLayer, idInUse := r.byid[id]; idInUse {
```

Comment:
(I know this is not something that you've added but) should a map access be protected with a mutex?

Reply:
`*layerStore` is proetected by `*layerStore.Lock()` and `*layerStore.Unlock()`.